plugins/aws/resoto_plugin_aws/resource/dynamodb.py

```diff
@@ -0,0 +1,377 @@
+
+    def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
+        super().connect_in_graph(builder, source)
+        if self.dynamodb_latest_stream_arn:
+            builder.dependant_node(
+                self,
+                clazz=AwsKinesisStream,
+                arn=self.dynamodb_latest_stream_arn,
+            )
+
+    def update_resource_tag(self, client: AwsClient, key: str, value: str) -> bool:
+        client.call(
+            service=self.api_spec.service,
+            action="tag-resource",
+            result_name=None,
+            ResourceArn=self.arn,
+            Tags=[{"Key": key, "Value": value}],
+        )
+        return True
+
+    def delete_resource_tag(self, client: AwsClient, key: str) -> bool:
+        client.call(
+            service=self.api_spec.service,
+            action="untag-resource",
+            result_name=None,
+            ResourceArn=self.arn,
+            TagKeys=[key],
+        )
+        return True
+
+    def delete_resource(self, client: AwsClient) -> bool:
+        client.call(service=self.api_spec.service, action="delete-table", result_name=None, TableName=self.name)
+        return True
+
+
+@define(eq=False, slots=False)
+class AwsDynamoDbGlobalTable(AwsResource):
+    kind: ClassVar[str] = "aws_dynamo_db_global_table"
+    api_spec: ClassVar[AwsApiSpec] = AwsApiSpec("dynamodb", "list-global-tables", "GlobalTables")
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("GlobalTableName"),
+        "name": S("GlobalTableName"),
+        "ctime": S("CreationDateTime"),
+        "arn": S("GlobalTableArn"),
+        "dynamodb_replication_group": S("ReplicationGroup", default=[])
+        >> ForallBend(AwsDynamoDbReplicaDescription.mapping),
+        "dynamodb_global_table_status": S("GlobalTableStatus"),
+    }
+    arn: Optional[str] = field(default=None)
+    dynamodb_replication_group: List[AwsDynamoDbReplicaDescription] = field(factory=list)
+    dynamodb_global_table_status: Optional[str] = field(default=None)
+
+    @classmethod
+    def collect(cls: Type[AwsResource], json: List[Json], builder: GraphBuilder) -> None:
+        def add_instance(table: Dict[str, str]) -> None:
+            table_description = builder.client.call(
+                "dynamodb", "describe-global-table", "GlobalTableDescription", GlobalTableName=table["GlobalTableName"]
+            )
+            instance = cls.from_api(table_description)
+            builder.add_node(instance, table_description)
+            builder.submit_work(add_tags, instance)
+
+        def add_tags(table: AwsDynamoDbGlobalTable) -> None:
+            tags = builder.client.list("dynamodb", "list-tags-of-resource", "Tags", ResourceArn=table.arn)
+            if tags:
+                table.tags = bend(ToDict(), tags)
+
+        for table in json:
+            builder.submit_work(add_instance, table)
+
+    def update_resource_tag(self, client: AwsClient, key: str, value: str) -> bool:
```

Comment:
WDYT about moving these methods into a class (since both resources can be tagged in exactly the same way)?

Reply:
I made an attempt - can you take a look?